pkgs/top-level/all-packages.nix

```diff
@@ -1146,17 +1147,18 @@ with pkgs;
             fetchurl = stdenv.fetchurlBoot;
           };
         });
-        perl = buildPackages.perl.override { fetchurl = stdenv.fetchurlBoot; };
+        perl = buildPackages.perl.override { inherit zlib; fetchurl = stdenv.fetchurlBoot; };
         openssl = buildPackages.openssl.override {
           fetchurl = stdenv.fetchurlBoot;
           buildPackages = {
             coreutils = buildPackages.coreutils.override {
               fetchurl = stdenv.fetchurlBoot;
               inherit perl;
               xz = buildPackages.xz.override { fetchurl = stdenv.fetchurlBoot; };
-              gmp = null;
+              gmpSupport = false;
               aclSupport = false;
               attrSupport = false;
+              autoreconfHook = null;
```

Comment:
Setting inputs to null like this can be very confusing, and is something we're trying to phase out.  There's a comment that says it's only needed here on Cygwin, so maybe we could only conditionally add it to nativeBuildInputs in that case.